lib/_http_agent.js

```diff
@@ -116,6 +117,24 @@ function Agent(options) {
       }
     }
   });
+
+  // Don't emit keylog events unless there is a listener for them.
+  const enableKeylog = (eventName) => {
+    if (eventName === 'keylog') {
+      this.removeListener('newListener', enableKeylog);
```

Comment:
You should match this with a `'removeListener'` listener in case the `'keylog'` event listener is removed again. Basically, you need to keep a listener count.

Also, the tests should verify that it works. :-)

Reply:
I don't think we should refcount  the keylog event listeners. Its not done in the other places that somewhat expensive features are  turned on only when needed. Its true that a node instance that listened on keylog, then stopped listening on keylog, would still be paying the price of keylog events emitted and ignored, but that strikes me as unusual, and a case of "don't do that".